src/channel.c

```diff
@@ -2222,6 +2225,9 @@ libssh2_channel_write_ex(LIBSSH2_CHANNEL *channel, int stream_id,
     if(!channel)
         return LIBSSH2_ERROR_BAD_USE;
 
+    if(!channel->session)
+        return LIBSSH2_ERROR_SOCKET_RECV;
```

Comment:
Can you explain the reason for this check? As far as I understand, there should never be a channel with a NULL session.

Also, the choice of error code here seems odd. Why return this error code?

Reply:
This check is unrelated with this patch and was wrongly pushed. I've stomped with channel->session == NULL and put this check to came along without crash, but I do not know why this error happened. I got the same impression that is an impossible situation, but I do not bothered to go deeper.

Reply:
Unless I'm misunderstanding something, should we delete this check if it isn't related to this PR, or is this something we need?